### PR TITLE
Make Repeater name configurable

### PR DESCRIPTION
**Pull request by SmittieC**
## Technical Summary
See the [ticket](https://dimagi-dev.atlassian.net/jira/software/c/projects/SC/boards/196?modal=detail&selectedIssue=SC-2361) for the reason for this PR. (This PR was reviewed [here](https://github.com/dimagi/commcare-hq/pull/32365), but keeping it up to date have proven to be a mission. It was easier to just create a new branch and port the changes than to kry and resolve the conflicts)

This is the third and last PR in a series of 3 that aims to add a name field to the Repeaters.
Quick summary: Now that the name attribute exists on the Repeater model, we can update it. This PR adds code that exposes a `name` field when the user creates a new Repeater or update an existing one. This field allows the user to either
1. Give the repeater a custom name
2. if left blank, the repeater name will default to its connection setting's name

See the screenshot below of what this new form looks like:
## When creating a repeater:
Screenshot from 2022-11-28 14-23-59

###
We can now see the new name (and it's not defaulting to "Connection 1")
Screenshot from 2022-11-28 14-25-19

## When updating a repeater:
Screenshot from 2022-11-28 14-25-46

This PR can only be merged once the [first PR](https://github.com/dimagi/commcare-hq/pull/32359) is deployed

## Safety Assurance
- I tested adding, updating and removing repeaters locally and on Staging.
- Tests pass (after failing and fixing)

### Safety story
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by mkangia — COMMENTED** [review]
Great documentation in the PR & code @SmittieC, made it easy to read and understand.
All looks good to me. I have not been following previous PRs so can't drop a review, but can do a proper review if needed.

File: corehq/motech/repeaters/forms.py [hunk 1]
```diff
@@ -21,6 +21,11 @@
 class GenericRepeaterForm(forms.Form):
 
     def __init__(self, *args, **kwargs):
+        self.repeater_name = ""
+        data = kwargs.get('data')
+        if data:
```
Comment (sravfeyn) [inline corehq/motech/repeaters/forms.py:26]:
```suggestion
        if data in kwargs:
```

File: corehq/motech/repeaters/views/repeaters.py [hunk 2]
```diff
@@ -141,6 +141,11 @@ def set_repeater_attr(self, repeater, cleaned_data):
         repeater.connection_settings_id = int(cleaned_data['connection_settings_id'])
         repeater.request_method = cleaned_data['request_method']
         repeater.format = cleaned_data['format']
+        name = cleaned_data.get('name')
+        if not name:
+            conn_settings = ConnectionSettings.objects.get(pk=repeater.connection_settings_id)
```
Comment (sravfeyn) [inline corehq/motech/repeaters/views/repeaters.py:146]: Should this catch a DoesNotExist error?

Comment (SmittieC) [reply corehq/motech/repeaters/views/repeaters.py:146]: There's a `NOT NULL` constraint on the `connection_settings_id` field on the repeaters.


Screenshot from 2023-01-18 10-25-19

**Review by sravfeyn — APPROVED** [review]

File: corehq/motech/repeaters/forms.py (hunk 1 above)
Comment (SmittieC) [reply corehq/motech/repeaters/forms.py:26]: The next line (in the if branch) needs this data variable.
```
self.repeater_name = data.get("name")
```

**Review by AmitPhulera — APPROVED** [review]
Looking good 🎉

**Pull request merged**